# Generated by Django 5.0.6 on 2025-04-11 12:08

import django.core.validators
from django.db import migrations, models


class Migration(migrations.Migration):

    dependencies = [
        ('eateries', '0012_productimage'),
    ]

    operations = [
        migrations.CreateModel(
            name='UserProfile',
            fields=[
                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('created_at', models.DateTimeField(auto_now_add=True)),
                ('updated_at', models.DateTimeField(auto_now=True)),
                ('type', models.CharField(choices=[('manager', 'Meneger'), ('customer', 'Mijoz')], default='customer', max_length=10)),
                ('full_name', models.CharField(blank=True, max_length=255, null=True, verbose_name='Full name')),
                ('phone_number', models.CharField(max_length=13, unique=True, validators=[django.core.validators.RegexValidator(message='Invalid phone number.', regex='^\\+?\\d{1,4}[\\s-]?\\(?\\d{1,4}\\)?[\\s-]?\\d{1,12}([\\s-]?\\d{1,12})?$')], verbose_name='Phone number')),
                ('telegram_id', models.CharField(blank=True, max_length=20, null=True, verbose_name='Telegram ID')),
                ('password', models.CharField(max_length=255)),
            ],
            options={
                'verbose_name': 'User profile',
                'verbose_name_plural': 'User profiles',
                'ordering': ['-created_at'],
            },
        ),
    ]
